src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcherImpl.java

```diff
@@ -141,7 +141,7 @@ public void listTasksInDB() throws SQLException, ParseException {
     }
 
     @Override
-    public List<Task> fillDB(SubmissionParameters submissionParameters) {
+    public List<Task> fillDB(SubmissionParameters submissionParameters, List<Date> datesToExclude) {
```

Comment:
this alg. is complicated. could you add some internal comments explaining it?

Reply:
Of course. Besides the comments, you think is better to rename the list `datesToExclude` to `alreadyProcessedDates` or just `processedDates`?

Because these dates are the dates from tasks recovered from others SAPS instances, that is, they don't need to be processed locally.